### PR TITLE
Port the project to C++11 - Part 3

### PR DESCRIPTION
**Pull request by lzaoral**
See commit messages.

### CODE REVIEW

File: src/lib/msg-filter.hh [hunk 1]
```diff
@@ -32,12 +33,13 @@ using TSubstMap = std::map<std::string, std::string>;
 class MsgFilter {
     public:
         // singleton
-        static MsgFilter* inst() {
-            return (self_)
-                ? (self_)
-                : (self_ = new MsgFilter);
+        static MsgFilter &inst() {
```
Comment (kdudka) [inline src/lib/msg-filter.hh:36]: I would prefer to spell this as `static MsgFilter& inst()` as we do in other function declarations.

File: src/lib/msg-filter.hh [hunk 2]
```diff
@@ -47,18 +49,19 @@ class MsgFilter {
...
-        ~MsgFilter();
 
-        bool setJSONFilter(InStream &filter);
+        bool setJSONFilter(InStream &input);
 
-        static MsgFilter *self_;
         struct Private;
-        Private *d;
+        std::unique_ptr<Private> d;
+        static std::unique_ptr<MsgFilter> self_;
```
Comment (kdudka) [inline src/lib/msg-filter.hh:64]: I am not sure whether we want to use `std::unique_ptr` for the global variable.  It introduces one more thing to debug on program exit.  Is the motivation behind the change to have clean valgrind output?

File: src/lib/csv-parser.cc [hunk 3]
```diff
@@ -34,10 +34,7 @@ AbstractCsvParser::AbstractCsvParser():
 {
 }
 
-AbstractCsvParser::~AbstractCsvParser()
-{
-    delete d;
-}
+AbstractCsvParser::~AbstractCsvParser() = default;
```
Comment (kdudka) [inline src/lib/csv-parser.cc:37]: I propose to squash commits 0b773b27f3e916cba754c146d2c57ff92ed76806 and fc483a65105ca1d636cde9965608f3e452ee3e95.  As it is now, the second commit fixes a memory leak that the first one introduces.

**Review by kdudka — APPROVED** [review]
Looks good. Minor comments inline...

File: src/lib/msg-filter.hh (hunk 2 above)
Comment (lzaoral) [reply src/lib/msg-filter.hh:64]: The main motivation was consistency. But I have no strong feelings about this one. Should I remove this change?

File: src/lib/msg-filter.hh (hunk 1 above)
Comment (lzaoral) [reply src/lib/msg-filter.hh]: Addressed in 78b43f890cbd4fcd2b7843a519f0fe2f9cffa758.

File: src/lib/csv-parser.cc (hunk 3 above)
Comment (lzaoral) [reply src/lib/csv-parser.cc:37]: Thanks! I've mixed those two by a mistake. Addressed in a92210518f357a3a2a4f7aa6b634452a5a556c3e and e419c57a39b15090d71d3be6619d197665ae996e.

File: src/lib/msg-filter.hh (hunk 2 above)
Comment (kdudka) [reply src/lib/msg-filter.hh:63]: Yes, please.  I would prefer to use a raw pointer for the static variable.

**Pull request merged**